File: frontend/types/agent.ts
export interface Agent {
  id: number;
  name: string;
  role: string;
  goal: string;
  backstory: string;
  tools: any[];
  llm_config: Record<string, any>;
  additional_params: Record<string, any>;
  created_at: string;
  updated_at: string;
}

export interface AgentCreate {
  name: string;
  role: string;
  goal: string;
  backstory: string;
  tools?: any[];
  llm_config?: Record<string, any>;
  additional_params?: Record<string, any>;
}

export interface AgentUpdate {
  name?: string;
  role?: string;
  goal?: string;
  backstory?: string;
  tools?: any[];
  llm_config?: Record<string, any>;
  additional_params?: Record<string, any>;
}

export interface AgentFilters {
  search?: string;
  skip?: number;
  limit?: number;
} 

File: backend/app/models/agent.py
from sqlalchemy import Column, Integer, String, Text, DateTime, JSON
from sqlalchemy.sql import func
from app.core.database import Base

class Agent(Base):
    __tablename__ = "agents"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False)
    role = Column(Text, nullable=False)
    goal = Column(Text, nullable=False)
    backstory = Column(Text, nullable=False)
    tools = Column(JSON, default=[])
    llm_config = Column(JSON, default={})
    additional_params = Column(JSON, default={})
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now()) 